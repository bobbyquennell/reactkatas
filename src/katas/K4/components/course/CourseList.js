import React from 'react';
import {connect} from 'react-redux';
import * as courseActions from '../../actions/courseActions';
class CourseList extends React.Component{
    constructor(props){
        super(props);
        this.state ={
            courseName: ""
            // courseList:[]
        };
    }

    submitHandler = (event)=>{
        event.preventDefault();
        //alert(`adding course: ${this.state.courseName}`);
        /* 1) add course to course List in a react way, without Redux*/
        // this.setState((prevState)=>{
        //     return({
        //         // courseList: [...prevState.courseList,
        //         //     this.state.courseName],
        //         courseList: prevState.courseList.concat(this.state.courseName),
        //         courseName: ""
        //     });
        // });

        /* 2) add course to course List in a redux way A: the most verbose way*/
        //this.props.dispatch(courseActions.createCourse(this.state.courseName));
        /* 3) add course to course list in a redux way B: utilize mapDispatchToProps function*/
        this.props.createCourse(this.state.courseName);
        this.setState({
            courseName: ""
        });
    }
    onChangeHandler = (event)=>{
        this.setState({
            courseName: event.target.value
        });
    }
    courseRow(title, index){
        return <div key={index}>{title}</div>;
    }

    render(){
        console.log(this.props.courses);
        return(

            <div>
                <h2>CourseList Page</h2>
                <form onSubmit={this.submitHandler}>
                    <input type="text" placeholder="course name" value={this.state.courseName} onChange={this.onChangeHandler} required/>
                    <button type="submit">Add Course</button>
                </form>
                <h4>course list:</h4>
                {this.props.courseList.map(this.courseRow)}
            </div>
        );
    }

}

// using react-redux connect function to connect your component to the redux store
//see: https://github.com/reactjs/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options
function mapStateToProps(state){
    return{
        courseList: state
    };
}
function mapDispatchToProps(dispatch){// the function's name can be anything, does not have to be mapDispatchToProps
    return{
        createCourse: course=>dispatch(courseActions.createCourse(course))
    };
}
export default connect(mapStateToProps, mapDispatchToProps)(CourseList);
