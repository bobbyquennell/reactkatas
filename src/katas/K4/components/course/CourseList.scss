@import "../../theme";

.courseButton {
  background-color: $white;
  display: block;
  margin: auto;
  width: 50%;
}
