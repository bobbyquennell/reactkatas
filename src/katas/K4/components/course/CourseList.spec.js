import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { CourseList } from './CourseList';

describe('CourseList', ()=>{
  it('should render correctly', ()=>{
    const props = {
      courseList: [],
      lecturers:[],
      asyncInProgress: 0
    };
    const wrapper = shallow(
      <CourseList {...props}/>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
