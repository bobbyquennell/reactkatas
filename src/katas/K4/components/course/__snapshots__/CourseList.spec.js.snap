// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseList: When async operation is running should render loading dots 1`] = `
<CourseList
  asyncInProgress={0}
  courseList={Array []}
  lecturers={Array []}
>
  <div>
    <Button
      name="Create Course"
      onClick={[Function]}
    >
      <button
        className=""
        onClick={[Function]}
      >
        Create Course
      </button>
    </Button>
    <CourseTable
      courses={Array []}
      lecturers={Array []}
    >
      <table
        style={
          Object {
            "margin": "20px auto",
            "maxWidth": "90%",
          }
        }
      >
        <tbody>
          <tr>
            <th
              style={
                Object {
                  "textAlign": "center",
                  "width": "20%",
                }
              }
            >
              Title
            </th>
            <th
              style={
                Object {
                  "textAlign": "center",
                  "width": "20%",
                }
              }
            >
              Lecturer
            </th>
            <th
              style={
                Object {
                  "textAlign": "center",
                  "width": "20%",
                }
              }
            >
              Address
            </th>
            <th
              style={
                Object {
                  "textAlign": "center",
                  "width": "20%",
                }
              }
            >
              Date
            </th>
            <th
              style={
                Object {
                  "textAlign": "center",
                  "width": "20%",
                }
              }
            >
              Time
            </th>
          </tr>
        </tbody>
      </table>
    </CourseTable>
     
  </div>
</CourseList>
`;
